### PR TITLE
adding version 3.4.20221105

### PR DESCRIPTION
**Pull request by erinyoung**
I copied everything for Cecret version 3.4.20221105 and then added the relevant metadata (I think).

### CODE REVIEW

**Comment by erinyoung** [comment]
This version basically has the same inputs as the prior version. 

MPX uses a different reference and primer sequence.

File: UPHL-BioNGS_Cecret_V3.4/main.nf [hunk 1]
```diff
@@ -0,0 +1,489 @@
...
+
+if ( params.fastas == params.multifastas ) {
+  println("'params.fastas' and 'params.multifastas' cannot point to the same directory!")
+  println("'params.fastas' is set to " + params.fastas)
+  println("'params.multifastas' is set to " + params.multifastas)
+  exit 1
+}
+
+//# outdir params
+params.outdir                               = workflow.launchDir + '/cecret'
```
Comment (jwarnn) [inline UPHL-BioNGS_Cecret_V3.4/main.nf]: ICA requires the outdir to be 'out'

```
params.outdir                               = workflow.launchDir + 'out/cecret'
```

File: UPHL-BioNGS_Cecret_V3.4/main.nf [hunk 2]
```diff
@@ -0,0 +1,489 @@
...
+params.samtools_plot_ampliconstats_options  = '-size 1200,900 -size2 1200,900 -size3 1200,900'
+params.samtools_markdup_options             = ''
+params.samtools_fixmate_options             = ''
+params.mafft_options                        = '--maxambiguous 0.5'
+params.snpdists_options                     = '-c'
+params.iqtree2_options                      = '-ninit 2 -n 2 -me 0.05 -m GTR'
+params.multiqc_options                      = ''
+
+//# for optional contamination determination
+params.kraken2_db                           = false
```
Comment (jwarnn) [inline UPHL-BioNGS_Cecret_V3.4/main.nf]: ```params.kraken2_db                 = workflow.launchDir + '/kraken2_db/minikraken2_v2_8GB_201904_UPDATE'```
I would suggest hard coding the kraken2_db here. The other option would be to add it to the xml but I haven't tested that and this was trick parameter as it was.

**Review by jwarnn — CHANGES_REQUESTED** [review]
I looked over the nf.main, xml config, and all of the modules files. Except for a few changes needed in nf.main it looks good. I would also suggest deleting some files that are not needed and would only take up space on ICA; I am mostly just concerned about the image files. I would like to run a test run using these files on ICA before merging this PR but will have a hard time fitting that in as I think Clarity UAT testing, and on-boaring Phoenix should take priority for me. If you could make those changes, hopefully I could test it out sometime next week.

File: UPHL-BioNGS_Cecret_V3.4/main.nf (hunk 1 above)
Comment (erinyoung) [reply UPHL-BioNGS_Cecret_V3.4/main.nf]: Good to know! I'll change all of them.

File: UPHL-BioNGS_Cecret_V3.4/main.nf (hunk 2 above)
Comment (erinyoung) [reply UPHL-BioNGS_Cecret_V3.4/main.nf]: This line isn't changed in Cecret V3. Do you have an example that I could look at for this?

**Comment by erinyoung** [comment]
I don't think I can hardcode the reference, gff, primer scheme, or amplicon files for this. These files change depending on whether we're doing MPX or COVID. I think it'd be fancy if the options were dropdown menus, but I'm not sure how to specify that with the xml file.

**Comment by erinyoung** [comment]
> and on-boaring Phoenix should take priority for me

Yeah... but there's actually a new version coming out soon.

File: UPHL-BioNGS_Cecret_V3.4/main.nf (hunk 2 above)
Comment (jwarnn) [reply UPHL-BioNGS_Cecret_V3.4/main.nf]: https://github.com/UPHL-BioNGS/ICA_Modified_Nextflow_Scripts/blob/main/UPHL-BioNGS_Granduer_V2/main.nf

I don't think I ever ran a dsl 2 version of Cecert on ICA; I was looking and could not find the github repository on ICA so  that might be why these weren't correct in the UPHL-BioNGS_Cecret_V3 version that is currently in the repo.

**Comment by jwarnn** [comment]
Since the the reference, gff, primer scheme, and  amplicon files are in the Cecret/configs/ folder they don't need to be hard coded. But we are bringing in our own database for kraken2 and we need to hard code in at least ```workflow.launchDir + '/kraken2_db/```  or maybe ``` ../kraken2_db/ ``` because the database will be loaded in the ../data folder of the ICA AWS instance that Nextflow is ran in. We could use the xml configuration and  would probably need to add compound parameters to the Nextflow script so that you could use different databases but I would need to work on that.

I put this in a comment further up but I want to repost to make sure it is seen. "I don't think I ever ran a dsl 2 version of Cecert on ICA; I was looking and could not find the github repository on ICA so that might be why these weren't correct in the UPHL-BioNGS_Cecret_V3 version that is currently in the repo."

File: UPHL-BioNGS_Cecret_V3.4/main.nf (hunk 2 above)
Comment (erinyoung) [reply UPHL-BioNGS_Cecret_V3.4/main.nf]: Perfect! Thank you!

**Comment by erinyoung** [comment]
Alright. I think I've made the relevant changes. Let me know if there's anything else.

(I don't have a deadline for you for this. It's just easier for me to work on things while they're in my mind.)

**Comment by jwarnn** [comment]
I am going to change this PR into updating to 3.5.20230201. I Will go ahead and make the needed changes.

**Comment by jwarnn** [comment]
I am going to close this and start a fresh one. Just easier that way.

**Pull request closed without merging**